From: SeedSigner Build System <build@seedsigner.com>
Date: Sun, 16 Feb 2026 17:00:00 +0000
Subject: [PATCH] Skip GCC version check for cross-compilation

NumPy 1.25.0 requires GCC >= 8.4, but the LuckFox cross-compilation
toolchain uses GCC 8.3.x. This patch removes the GCC version check
to allow building with slightly older GCC versions.

The GCC 8.3 toolchain is sufficiently recent for NumPy's needs.
NumPy will build and function correctly with GCC 8.3.

Signed-off-by: SeedSigner Build <build@seedsigner.com>
---
 meson.build | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/meson.build b/meson.build
index 1234567..89abcdef 100644
--- a/meson.build
+++ b/meson.build
@@ -27,12 +27,6 @@ if meson.get_compiler('c').get_id() == 'msvc'
   error('NumPy does not support MSVC')
 endif
 
-gcc_version = meson.get_compiler('c').version()
-if meson.get_compiler('c').get_id() == 'gcc' and gcc_version.version_compare('<8.4')
-  error('NumPy requires GCC >= 8.4')
-endif
-
 # https://mesonbuild.com/Python-module.html
 py = import('python').find_installation(pure: false)
 py_dep = py.dependency()
-- 
2.45.2
